cache.go

```diff
@@ -647,15 +645,26 @@ func cacheHandler(next http.HandlerFunc, w http.ResponseWriter, r *http.Request,
 			for k, v := range result.Header {
 				w.Header().Set(k, strings.Join(v, ","))
 			}
-			value := rec.Body.Bytes()
+			buf := make([]byte, 1024*1024)
+			var cacheBuf bytes.Buffer
+			length := 0
+			for {
+				n, err := rec.Body.Read(buf)
+				if err == io.EOF || n == 0 {
+					break
+				}
+				length += n
+				w.Write(buf[:n])
+				cacheBuf.Write(buf[:n])
```

Comment:
storing object content in memory will cause sidekick to crash if the size of the object is large, it needs to be streamed to the cache

Reply:
fixed.

Reply:
it's still cached in memory:
```
20type ResponseRecorder struct {                                                                                                                                                                                  
 21        // Code is the HTTP response code set by WriteHeader.                                                                                                                                                   
 22        //                                                                                                                                                                                                      
 23        // Note that if a Handler never calls WriteHeader or Write,                                                                                                                                             
 24        // this might end up being 0, rather than the implicit                                                                                                                                                  
 25        // http.StatusOK. To get the implicit value, use the Result                                                                                                                                             
 26        // method.                                                                                                                                                                                              
 27        Code int                                                                                                                                                                                                
 28                                                                                                                                                                                                                
 29        // HeaderMap contains the headers explicitly set by the Handler.                                                                                                                                        
 30        // It is an internal detail.                                                                                                                                                                            
 31        //                                                                                                                                                                                                      
 32        // Deprecated: HeaderMap exists for historical compatibility                                                                                                                                            
 33        // and should not be used. To access the headers returned by a handler,                                                                                                                                 
 34        // use the Response.Header map as returned by the Result method.                                                                                                                                        
 35        HeaderMap http.Header                                                                                                                                                                                   
 36                                                                                                                                                                                                                
 37        // Body is the buffer to which the Handler's Write calls are sent.                                                                                                                                      
 38        // If nil, the Writes are silently discarded.                                                                                                                                                           
 39        Body *bytes.Buffer                                                                                                                                                                                      
```

```
106func (rw *ResponseRecorder) Write(buf []byte) (int, error) {                                                                                                                                                    
107        rw.writeHeader(buf, "")                                                                                                                                                                                 
108        if rw.Body != nil {                                                                                                                                                                                     
109                rw.Body.Write(buf)                                                                                                                                                                              
110        }                                                                                                                                                                                                       
111        return len(buf), nil                                                                                                                                                                                    
112}                                                                                                                                                                                                               
```